src/aspire/classification/averager2d.py

```diff
@@ -0,0 +1,1018 @@
+import logging
+from abc import ABC, abstractmethod
+from itertools import product
+
+import matplotlib.pyplot as plt
+import numpy as np
+from skimage.filters import difference_of_gaussians, window
+from skimage.transform import rotate, warp_polar
+from tqdm import tqdm, trange
+
+from aspire.image import Image
+from aspire.numeric import fft
+from aspire.source import ArrayImageSource
+from aspire.utils.coor_trans import grid_2d
+
+logger = logging.getLogger(__name__)
+
+
+class Averager2D(ABC):
+    """
+    Base class for 2D Image Averaging methods.
+    """
+
+    def __init__(self, composite_basis, source, batch_size=512, dtype=None):
+        """
+        :param composite_basis:  Basis to be used during class average composition (eg FFB2D)
+        :param source: Source of original images.
+        :param dtype: Numpy dtype to be used during alignment.
+        """
+
+        self.composite_basis = composite_basis
+        self.src = source
+        self.batch_size = batch_size
```

Comment:
This was put in to move toward the batching/`images` approach later, but you are right that it is not used.

It could def make sense for some of the `averager`...  if you have estimated rotations and shifts you can go over data once... (at one point I did that in the draft code).  it maybe doesn't make sense for the `aligner`s.

Anyway, future problems, Rm'd.